ocs_ci/ocs/platform_nodes.py

```diff
@@ -0,0 +1,276 @@
+import logging
+
+from ocs_ci.ocs.exceptions import TimeoutExpiredError
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework import config
+from ocs_ci.utility import aws
+from ocs_ci.ocs.node import get_node_objs
+from ocs_ci.ocs.resources.pvc import get_deviceset_pvs
+
+
+logger = logging.getLogger(__name__)
+
+
+class PlatformNodesFactory:
+    """
+    A factory class to get specific nodes platform object
+
+    """
+    def __init__(self):
+        self.cls_map = {'AWS': AWSNodes, 'VMWare': VMWareNodes}
+
+    def get_nodes_platform(self):
+        platform = config.ENV_DATA['platform']
+        return self.cls_map[platform]()
+
+
+class NodesBase(object):
```

Comment:
I'm just curious, wouldn't be better to rise the `NotImplementedError` for the methods in this `NodeBase` class instead of in the child classes?